bdb/bdb_verify.c

```diff
@@ -521,564 +798,451 @@ static int bdb_verify_ll(
+    for (int ix = 0; ix < par->bdb_state->numix; ix++) {
+        rc = bdb_verify_key(par, ix, lid);
+        if (rc)
+            goto done;
+    }
 
-            if (keylen != dbt_key.size) {
-                ret = 1;
-                locprint(sb, lua_callback, lua_params,
-                    "!%016llx ix %d key size mismatch expected %d got %d\n",
-                    genid_flipped, ix, keylen, dbt_key.size);
-                goto next_key;
-            }
+    /* scan 3: scan each blob, verify data exists */
+    int nblobs = get_numblobs(par->db_table);
+    for (int blobno = 0; blobno < nblobs; blobno++) {
+        for (int dtastripe = 0; dtastripe < par->bdb_state->attr->blobstripe;
+             dtastripe++) {
+            bdb_verify_blob(par, blobno, dtastripe, lid);
+        }
+    }
 
-            unsigned long long genid_left, genid_right, masked_genid;
-
-            if (bdb_state->ixdta[ix]) {
-                /*  if dtacopy, does data payload in the key match the data
-                 * payload in the dta file? */
-                int expected_size;
-                uint8_t *expected_data;
-                uint8_t datacopy_buffer[bdb_state->lrl];
-                if (bdb_state->datacopy_odh) {
-                    int odhlen;
-                    unpack_index_odh(bdb_state, &dbt_data, &genid_right,
-                                     datacopy_buffer, sizeof(datacopy_buffer),
-                                     &odhlen, &ver);
-                    expected_size = odhlen;
-                    vtag_callback(callback_parm, datacopy_buffer,
-                                  &expected_size, ver);
-                    expected_data = datacopy_buffer;
-                } else {
-                    expected_size = dbt_data.size - sizeof(genid);
-                    expected_data = (uint8_t *)dbt_data.data + sizeof(genid);
-                    memcpy(&genid_right, (uint8_t *)dbt_data.data,
-                           sizeof(genid));
-                }
+done:
+    return par->verify_status;
+}
 
-                if (expected_size != bdb_state->lrl) {
-                    ret = 1;
-                    locprint(sb, lua_callback, lua_params, "!%016llx ix %d dtacpy payload wrong size "
-                                    "expected %d got %d\n",
-                                genid_flipped, ix, bdb_state->lrl,
-                                expected_size);
-                    goto next_key;
-                }
+/* This function serves as a wrapper around calling individual functions
+ * to verify data, key, and blob. 
+ * It gets a separate locker id under which it processes the appropriate verify.
+ */
+void bdb_verify_handler(td_processing_info_t *info) 
+{
+    verify_common_t *par = info->common_params;
+    bdb_state_type *bdb_state = par->bdb_state;
+    int rc;
+    unsigned int lid;
+    BDB_READLOCK("bdb_verify");
+    
+    if ((rc = bdb_state->dbenv->lock_id_flags(bdb_state->dbenv, &lid,
```

Comment:
When you run this without parallel, this is a second lock you get in the same thread of control (tran was the first).

Reply:
This is the same as what we did before -- that logic did not change with this PR.